plugins/modules/dcnm_service_policy.py

```diff
@@ -0,0 +1,1962 @@
+          next_hop_option:  none
+          acl_name: fwd_acl_10
+          rev_acl_name: rev_acl_10
+          route_map_num: 101
+          rev_route_map_num: 102
+
+DELETE SERVICE POLICIES
+=======================
+
+1. With Policy Name and Node name
+
+Deletes the specific service policy specified from the given node
+
+- name: Delete service policies with policy name and node name
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: deleted
+    config:
+      - name: service_policy_1
+        node_name: it-sn-1
+            
+      - name: service_policy_2
+        node_name: it-sn-2
+            
+      - name: service_policy_3
+        node_name: it-sn-2
+            
+      - name: service_policy_4
+        node_name: it-sn-2
+              
+      - name: service_policy_5
+        node_name: it-sn-2
+
+2. With Node name alone
+
+Deletes all service policies from the specified nodes
+
+- name: Delete service policies with Node names
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: deleted
+    config:
+      - node_name: it-sn-1
+      - node_name: it-sn-2
+
+3. With Node name and RP name
+
+   Deletes all service policies under the specified route peering and node
+
+- name: Delete service policies with Node name and RP name
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: deleted
+    config:
+      - node_name: it-sn-1
+        rp_name: it-fw-rp1
+
+      - node_name: it-sn-2
+        rp_name: it-fw-rp2
+
+4. Without config
+
+   Deletes all service policies on the given fabric and attached fabirc
+
+- name: Delete service policies without config
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabirc
+    attached_fabric: external
+    state: deleted
+
+OVERRIDE SERVICE POLICIES
+=========================
+
+- name: Override and delete all service policies
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    state: overridden
+
+- name: Override all existing service policies with a new one
+  cisco.dcnm.dcnm_service_policy:
+    fabric: test_fabric
+    attached_fabric: external
+    attach: true
+    deploy: true
+    state: overridden
+    config:
+      - name: service_policy_1
+        node_name: it-sn-1
+        rp_name: it-fw-rp1
+        src_vrf: vrf_11
+        dest_vrf: vrf_11
+        src_network: net_11
+        dest_network: net_12
+        next_hop: 192.161.1.100
+        rev_next_hop: 192.161.2.100
+        policy:
+          proto: icmp
+          src_port: 555
+          dest_port: 22
+          action: permit
+          next_hop_option:  none
+          acl_name: fwd_acl_555
+          rev_acl_name: rev_acl_555
+          route_map_num: 555
+          rev_route_map_num: 556
+
+REPLACE SERVICE POLICIES
+========================
+
+- name: Replace some of the objects in already created service policy
```

Comment:
The way I read this it seems like replaced is more like a merge.  I would state that `service_policy_1` will be replaced on DCNM with the ansible playbook version.

Reply:
Done